etl/steps/__init__.py

```diff
@@ -73,22 +72,13 @@ def compile_steps(
 def to_dependency_order(
     dag: DAG, includes: List[str], excludes: List[str]
 ) -> List[str]:
-    # reverse the graph so that dependencies point to their dependents
-    graph = reverse_graph(dag)
-
-    if includes:
-        # cut the graph to just the listed steps and the things that
-        # then depend on them (transitive closure)
-        subgraph = filter_to_subgraph(graph, includes)
-
-        # make sure we also include all dependencies of the subgraph
-        nodes = set(subgraph.keys()) | {e for v in subgraph.values() for e in v}
-        subgraph = reverse_graph(maximal_subgraph(dag, nodes))
-    else:
-        subgraph = graph
-
-    # make sure dependencies are built before running the things that depend on them
-    in_order = topological_sort(subgraph)
+    """
+    Organize the steps in dependency order with a topological sort. In other words,
+    the resulting list of steps is a valid ordering of steps such that no step is run
+    before the steps it depends on. Note: this ordering is not necessarily unique.
+    """
+    subgraph = filter_to_subgraph(dag, includes) if includes else dag
+    in_order = list(graphlib.TopologicalSorter(subgraph).static_order())
```

Comment:
Inlining this, since it's shorter now without all the reversing + is the whole point of the `to_dependency_order` method